[CI] UpgradeClusterClientYamlTestSuiteIT class failing

**Build Scans:**
- [elasticsearch-periodic #7080 / 8.0.1_bwc](https://gradle-enterprise.elastic.co/s/sbaigwmwl7xva)
- [elasticsearch-periodic #7073 / 8.0.1_bwc](https://gradle-enterprise.elastic.co/s/qsfwtr5khwnq4)

**Reproduction Line:**
```
./gradlew ":x-pack:qa:rolling-upgrade:v8.0.1#oldClusterTest" -Dtests.class="org.elasticsearch.upgrades.UpgradeClusterClientYamlTestSuiteIT" -Dtests.method="test {p0=old_cluster/80_transform_jobs_crud/Test put batch transform on old cluster}" -Dtests.seed=F5377982D8D243B5 -Dtests.bwc=true -Dtests.locale=brx-IN -Dtests.timezone=Africa/Lagos -Druntime.java=24
```

**Applicable branches:**
8.18

**Reproduces locally?:**
Yes

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.upgrades.UpgradeClusterClientYamlTestSuiteIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(org.elasticsearch.upgrades.UpgradeClusterClientYamlTestSuiteIT),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: Failure in test setup: Failed to initialize at least 3 watcher nodes
```

**Issue Reasons:**
- [8.18] 2 failures in class org.elasticsearch.upgrades.UpgradeClusterClientYamlTestSuiteIT (0.2% fail rate in 1000 executions)
- [8.18] 2 failures in step 8.0.1_bwc (100.0% fail rate in 2 executions)
- [8.18] 2 failures in pipeline elasticsearch-periodic (100.0% fail rate in 2 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

I spent some time trying to reproduce this locally. The tests fail reliably locally for `:x-pack:qa:rolling-upgrade:v8.0.1#oldClusterTest`, including lots of logs about...
```
»  Caused by: java.lang.AssertionError: expected empty context but was {X-elastic-product-origin=fleet} on elasticsearch[v8.0.1-2][transport_worker][T#1]
»       at org.elasticsearch.transport.Transports.assertDefaultThreadContext(Transports.java:60) ~[elasticsearch-8.0.1.jar:8.0.1]
»       at org.elasticsearch.transport.netty4.Netty4MessageChannelHandler.write(Netty4MessageChannelHandler.java:91) ~[?:?]
»       at io.netty.channel.AbstractChannelHandlerContext.invokeWrite0(AbstractChannelHandlerContext.java:717) ~[?:?]
```
`:x-pack:qa:rolling-upgrade:v7.17.29#oldClusterTest` passes
`:x-pack:qa:rolling-upgrade:v8.0.0#oldClusterTest` fails (same assertion failures)
`:x-pack:qa:rolling-upgrade:v8.1.0#oldClusterTest` passes

I assessed it as low risk because the issue seems isolated to version 8.0.x, but could be convinced it's a medium given that it looks like a genuine BWC issue.

--- a/x-pack/qa/rolling-upgrade/src/test/java/org/elasticsearch/upgrades/SystemIndicesUpgradeIT.java
+++ b/x-pack/qa/rolling-upgrade/src/test/java/org/elasticsearch/upgrades/SystemIndicesUpgradeIT.java
@@ -40,6 +40,10 @@ public class SystemIndicesUpgradeIT extends AbstractUpgradeTestCase {
         new SecureString(SecuritySettingsSourceField.TEST_PASSWORD)
     );
 
+    public static void avoidBugIn8_0_1() {
+        assumeTrue("https://github.com/elastic/elasticsearch/issues/125167", isOriginalClusterVersionAtLeast(Version.V_8_1_0));
+    }
+
     @Override
     protected Settings restAdminSettings() {
         // Note that we are both superuser here and provide a product origin

--- a/x-pack/qa/rolling-upgrade/src/test/java/org/elasticsearch/upgrades/SystemIndicesUpgradeIT.java
+++ b/x-pack/qa/rolling-upgrade/src/test/java/org/elasticsearch/upgrades/SystemIndicesUpgradeIT.java
@@ -23,6 +23,7 @@ import org.elasticsearch.logging.Logger;
 import org.elasticsearch.tasks.Task;
 import org.elasticsearch.xcontent.json.JsonXContent;
 import org.elasticsearch.xpack.test.SecuritySettingsSourceField;
+import org.junit.BeforeClass;
 
 import java.io.IOException;
 import java.util.List;
@@ -40,6 +41,7 @@ public class SystemIndicesUpgradeIT extends AbstractUpgradeTestCase {
         new SecureString(SecuritySettingsSourceField.TEST_PASSWORD)
     );
 
+    @BeforeClass
     public static void avoidBugIn8_0_1() {
         assumeTrue("https://github.com/elastic/elasticsearch/issues/125167", isOriginalClusterVersionAtLeast(Version.V_8_1_0));
     }

